vllm/distributed/parallel_state.py

```diff
@@ -195,17 +202,25 @@ def initialize_model_parallel(
         )
 
     # Build the pipeline model-parallel groups.
-    global _PP_DEVICE_GROUP
+    global _PP_DEVICE_GROUP, _PP_CPU_GROUP
+    global _PP_PYNCCL_COMMUNICATOR
     global _PP_GLOBAL_RANKS
     assert _PP_DEVICE_GROUP is None, (
         "pipeline model parallel group is already initialized")
     for i in range(num_pipeline_model_parallel_groups):
         ranks = list(range(i, world_size, num_pipeline_model_parallel_groups))
         group = torch.distributed.new_group(ranks, backend=backend)
+        cpu_group = torch.distributed.new_group(ranks, backend="gloo")
         if rank in ranks:
             _PP_DEVICE_GROUP = group
+            _PP_CPU_GROUP = cpu_group
             _PP_GLOBAL_RANKS = ranks
 
+    _PP_PYNCCL_COMMUNICATOR = PyNcclCommunicator(
+        group=_PP_CPU_GROUP,
+        device=_LOCAL_RANK,
+    )
+
```

Comment:
creating communicators will add memory cost. please add a check here, if pp size == 1, skip the construction of `PyNcclCommunicator`.

Reply:
Changed both PP and TP.

Reply:
@youkaichao Do you know how much extra memory the communicators would be adding? It is a fixed cost or scales with group size? 

We are running into some OOM with cudagraph enabled that shouldn't be happening. Any pointers would be greatly appreciated!